src/utils/url.js

```diff
@@ -199,12 +199,17 @@ export function getHostUrl(url) {
 }
 
 /**
+ * Returns the canonical URL from the canonical tag. If the canonical tag is
+ * not present, treat the doc URL itself as canonical.
  * @param {!../model/doc.Doc} doc
  * @return {string}
  */
 export function getCanonicalUrl(doc) {
   const node = doc.getRootNode().querySelector("link[rel='canonical']");
```

Comment:
Looks like we call `doc.getRootNode()` 3x in this method. Would it make sense to call it once? Ex: `const rootNode = doc.getRootNode()`

Reply:
Totally makes sense. Revised accordingly.